### PR TITLE
[aoc-collector non-root user]: Privileged user access requirement with awscontainerinsightreceiver

### PR DESCRIPTION
**Pull request by PaurushGarg**
Adding documentation regarding using privileged root user access for `awscontainerinsightreceiver` once ADOT Collector image is released that runs with non-root `aoc` user. 

Dependent on PR: https://github.com/aws-observability/aws-otel-collector/pull/2297

Link to tracking Issue:
https://github.com/aws-observability/aws-otel-collector/issues/2222

### CODE REVIEW

File: src/docs/getting-started/container-insights.mdx [hunk 1]
```diff
@@ -32,6 +32,8 @@ CloudWatch Container Insights collects metrics for many resources such as CPU, m
 It also provides diagnostic information such as container restart failures.
 The metrics are aggregated at the cluster, node, pod, task, and service level as CloudWatch metrics.
 
+Note: Collecting infrastructure metrics through `awsontainerinsightreceiver` requires running ADOT Collector with privilidged root access as it also collects container-related data from `cadvisor`.
```
Comment (bryan-aguilar) [inline src/docs/getting-started/container-insights.mdx:35]: We should add an example somewhere on how to give the container elevated privileges in a k8s manifest.

File: src/docs/components/ecs-metrics-receiver.mdx [hunk 2]
```diff
@@ -32,6 +32,8 @@ To enable the `awsecscontainermetrics` receiver, add the name under receiver sec
 By default, the receiver scrapes the ECS task metadata endpoint every 20s and collects all metrics
 (For the full list of metrics, see [Available Metrics](#available-metrics)).
 
+Note: Collecting infrastructure metrics through `awsontainerinsightreceiver` requires running ADOT Collector with privilidged root access as it also collects container-related data from `cadvisor`.
```
Comment (rapphil) [inline src/docs/components/ecs-metrics-receiver.mdx:35]: why is this note in this document? doesn't this relate to `awscontainerinsightreceiver` only? this section is for `awsecscontainermetrics`

File: src/docs/getting-started/container-insights.mdx (hunk 1 above)
Comment (PaurushGarg) [reply src/docs/getting-started/container-insights.mdx]: Added.

File: src/docs/components/ecs-metrics-receiver.mdx (hunk 2 above)
Comment (PaurushGarg) [reply src/docs/components/ecs-metrics-receiver.mdx]: Removed.

File: src/docs/getting-started/container-insights.mdx [hunk 3]
```diff
@@ -32,6 +32,26 @@ CloudWatch Container Insights collects metrics for many resources such as CPU, m
...
+      securityContext:
+          runAsUser: 0
+          runAsGroup: 0
+```
+For ECS deployments, use the [Task Definition](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinition.html#cfn-ecs-taskdefinition-containerdefinition-user) to specify the user owner of the collector process:
+```
+{
+  "name": "aoc-collector",
+  "image": "public.ecr.aws/aws-observability/aws-otel-collector:latest",
+  "user": "0",
```
Comment (vasireddy99) [inline src/docs/getting-started/container-insights.mdx:50]:
```suggestion
  "user": "root",
```

File: src/docs/getting-started/container-insights/ecs-prometheus.mdx [hunk 4]
```diff
@@ -338,6 +339,7 @@ e.g. `cpu 255 memory 511` is invalid pair.
     {
       "name": "adot",
       "image": "amazon/aws-otel-collector:v0.11.0",
+      "user": "0",
```
Comment (vasireddy99) [inline src/docs/getting-started/container-insights/ecs-prometheus.mdx:342]:
```suggestion
      "user": "root",
```

File: src/docs/getting-started/container-insights.mdx (hunk 3 above)
Comment (PaurushGarg) [reply src/docs/getting-started/container-insights.mdx:50]: Vasi, I kept it as 0 as in the testing [here](https://github.com/aws-observability/aws-otel-test-framework/pull/1473) we are using 0. But I think both (0 and root) would do same thing. Let me know if you want me to still change it.

File: src/docs/getting-started/container-insights/ecs-prometheus.mdx (hunk 4 above)
Comment (PaurushGarg) [reply src/docs/getting-started/container-insights/ecs-prometheus.mdx:342]: Vasi, I kept it as 0 as in the testing [here](https://github.com/aws-observability/aws-otel-test-framework/pull/1473) we are using 0. But I think both (0 and root) would do same thing. Let me know if you want me to still change it.

**Review by vasireddy99 — APPROVED** [review]

**Pull request merged**